src/segment/segmentTemplate.js

```diff
@@ -273,7 +152,26 @@ export const segmentsFromTemplate = (attributes, segmentTimeline) => {
     RepresentationID: attributes.id,
     Bandwidth: parseInt(attributes.bandwidth || 0, 10)
   };
-  const mapUri = constructTemplateUrl(attributes.initialization || '', templateValues);
+
+  let mapSegment = { uri: '', resolvedUri: resolveUrl(attributes.baseUrl || '', '') };
+
+  if (attributes.initialization && typeof attributes.initialization === 'string') {
+    const mapUri = constructTemplateUrl(attributes.initialization || '', templateValues);
+
+    mapSegment = {
+      uri: mapUri,
+      resolvedUri: resolveUrl(attributes.baseUrl || '', mapUri)
+    };
+  }
+
+  if (attributes.initialization && typeof attributes.initialization === 'object') {
```

Comment:
Based on what I picked up off the spec, `<Initialization>` doesn't get templated, but `@initialization` does, which is why there is a difference here.